### PR TITLE
[travis-ci] Use separate build stage for pod lint

### PR DESCRIPTION
**Pull request by jessesquires**
https://docs.travis-ci.com/user/build-stages

https://docs.travis-ci.com/user/customizing-the-build/#build-matrix

https://docs.travis-ci.com/user/build-stages/matrix-expansion/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -65,3 +61,9 @@ script:
 
 after_success:
 - bash <(curl -s https://codecov.io/bash)
+
```
Comment (kevnm67) [inline .travis.yml:52]: 👍 I thought about asking your thoughts on stages last week.  
What are your thoughts on moving the tests to a stage?  
Example:

```yaml
- stage: test
  name: iPhone X 11.4 tests
  script: 
  - (destination test...)
- stage:
  name: iPhone X 12.1 tests
  script: 
  - (destination test...)
- stage:
  name: Apple TV 12.2 tests
  script: 
  - (destination test...)
```

**Comment by jessesquires** [comment]
@kevnm67 yes, i'd love to do that!

i was looking at the docs last night, and found them a bit confusing though.
- https://docs.travis-ci.com/user/customizing-the-build/#naming-jobs-within-matrices
- https://docs.travis-ci.com/user/build-stages/matrix-expansion/

I'd like to have:

- one stage for the tests
- one stage for building the example project (just ensures that library updates don't break the example)
- one stage for `pod lint` (done here)

**Comment by jessesquires** [comment]
@kevnm67 want to just collaborate on this PR and push some commits here?

**Comment by kevnm67** [comment]
@jessesquires Sounds good.
Their docs aren't great.  Ill push something tomorrow morning.

**Comment by kevnm67** [comment]
@jessesquires Take a look at [this build](https://travis-ci.com/kevnm67/JSQDataSourcesKit/builds/104554563).  [Here's the PR](https://github.com/kevnm67/JSQDataSourcesKit/pull/3) I testing the changes with.

- How do you want the tests to expand?
- General feedback

**Comment by jessesquires** [comment]
@kevnm67 nice! overall looks good. 👍 

- Not sure what the "JSON file test" is for. do we need that?
- For building the example, I'd like to build on all supported iOS versions

**Comment by jessesquires** [comment]
decided to go ahead and merge this. i think it'll be easier to follow-up with your changes separately 😄 

be sure to branch from `develop`

want to open a PR here for what you had?

**Comment by kevnm67** [comment]
@jessesquires My bad. I was playing around trying to fix the build errors on travis.  I should have separated those changes... 

Sounds good.  I'll open a PR with my original changes.

**Comment by jessesquires** [comment]
@kevnm67 all good! they had some intermittent problems, but they seem to be resolved now

**Pull request merged**